### DISCUSSION TITLE
Stop storing passwords in plaintext

### DISCUSSION THREAD

**Discussion by NotAFile** [piqueserver/piqueserver#682] [Feature Requests]
Passwords should be properly hashed/salted instead. If we do this we will want a handy utility to allow you to generate these hashes, and perhaps even insert them into your config, if a proper toml editing library exists now.

**Comment by godwhoa** [comment]
Let's go about it in a backwards compatible way:
```
if password.startswith("$2b$") and len(password) == 60:
    # bcrypt.check
else:
    # plain text compare
```
`piqueserver hash <password>` simple bcrypt utility for hashing

One worry is bcrypt's computation time causing event-loop stalls.

**Comment by NotAFile** [comment]
I think bcrypt is only available on 3.6+, with pbkdf being available on 3.5+

**Comment by godwhoa** [comment]
We could use a [bcrypt](https://pypi.org/project/bcrypt/) library, if you are against that I'm open to other ways of hashing.

**Comment by NotAFile** [comment]
sounds good. Although, for our purposes, the included pbkdf2 should be more than plenty anyway.
